src/lib/asn1/ASN1.h

```diff
@@ -188,10 +190,13 @@ class DLL_EXPORT ASN1Writer
     ASN1_ERROR PutValue(uint8_t cls, uint32_t tag, bool isConstructed, chip::TLV::TLVReader & val);
 
 private:
+    static constexpr size_t kMaxDeferredLengthDepth = kMaxConstructedAndEncapsulatedTypesDepth;
+
     uint8_t * mBuf;
     uint8_t * mBufEnd;
     uint8_t * mWritePoint;
-    uint8_t ** mDeferredLengthList;
+    uint8_t * mDeferredLength[kMaxDeferredLengthDepth];
```

Comment:
```suggestion
    uint8_t * mDeferredLengthLocations[kMaxDeferredLengthDepth];
```
And please document this member and the next one.